lib/ruby_lsp/ruby_document.rb

```diff
@@ -10,5 +10,48 @@ def parse
       @needs_parsing = false
       @parse_result = Prism.parse(@source)
     end
+
+    class SymbolHierarchyRoot
+      extend T::Sig
+
+      sig { returns(T::Array[Interface::DocumentSymbol]) }
+      attr_reader :children
+
+      sig { void }
+      def initialize
+        @children = T.let([], T::Array[Interface::DocumentSymbol])
+      end
+    end
+
+    class DocumentSymbolStack
```

Comment:
I know it's likely that this class probably won't be used outside of the document symbol feature. But I'm hesitant putting it under `Requests::DocumentSymbol` as I don't think addons should reference back to a specific request 🤔